clang/lib/Parse/ParseDeclCXX.cpp

```diff
@@ -2646,6 +2646,9 @@ bool Parser::ParseCXXMemberDeclaratorBeforeInitializer(
   else
     DeclaratorInfo.SetIdentifier(nullptr, Tok.getLocation());
 
+  if (getLangOpts().HLSL)
```

Comment:
It may be worth cleaning this up: I notice that there are two overloads for MaybeParseHLSLAnnotations.  One of them checks the language inside, and the other one only asserts the language mode.  Would it be better if these two overloads behaved the same way?

Reply:
I think the crucial difference between the two is that one overload is intended to parse annotations for the purpose of creating attributes to attach to a specific declarator. The other simply parses the present attributes and fills in a ParsedAttributes object. The latter is used in `ParseHLSLBuffer`, where there is no decl to attach the attributes to yet. The former is convenient since it will attach the attributes after parsing for you.
They could be combined if an extra parameter is passed, a pointer to a declarator, and then add the attributes if it's non-null, but I don't think that's really necessary.

Reply:
I'm not proposing we combine them.  My observation is that one of them will do the right thing if you call it outside of HLSL mode and the other one will do the wrong thing.  So, if you were to switch from using one overload to another you may find that you've introduce a surprising bug.

Reply:
Though both would hit an assert if they're called outside of HLSL mode, I do believe that one overload has a redundant check for HLSL mode, I'll remove that.